src/main/scala/is/hail/distributedmatrix/Keys.scala

```diff
@@ -0,0 +1,83 @@
+package is.hail.distributedmatrix
+
+import is.hail.annotations.Annotation
+import is.hail.expr.{JSONAnnotationImpex, Parser}
+import is.hail.expr.types.Type
+import is.hail.utils._
+import org.apache.spark.SparkContext
+import org.json4s.jackson.Serialization
+import org.json4s.{JArray, JValue}
+import org.json4s.jackson.JsonMethods.parse
+
+case class KeysImpex(typeStr: String, valuesJ: JValue)
```

Comment:
If valuesJ (why J?  The type is right there.) needs to be an array, you can make it `Array[JValue]` and extract will do the right thing.